### DISCUSSION TITLE
fixing typos in "survey-text" responses

### DISCUSSION THREAD

**Discussion by rbruno1** [jspsych/jsPsych#2024] [Q&A] [answered]
My experiment uses type: "survey-text". Is there a way to fix typo's in participants' responses? correct responses are being labelled as incorrect because of a space at the end or because of one missed letter.

**Comment by becky-gilbert** [comment, marked as answer]
Hi @rbruno1, you can easily get rid of the whitespace at the start and end of a string with the [JavaScript trim function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim). Detecting a missing letter is probably going to be trickier. You could look at using [regular expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions). One fairly simple thing you could do with regular expressions is remove any non-alphabetic characters from the response (assuming that the response should just be letters), which will catch typos containing numbers, punctuation etc.
